### PR TITLE
New command line switch to make it easier to load local programs

### PR DESCRIPTION
**Pull request by izb**
Add -mountdir CLI switch which changes the host directory after the ROM has loaded. Useful for when you have multiple versions of the emulator, but need to load programs or other files from a common location.

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -707,6 +720,16 @@ main(int argc, char **argv)
 		}
 	}
 
+	if (mountdir_path) {
+		DIR* dir = opendir(mountdir_path);
```
Comment (mist64) [inline main.c]: space, then asterisk.

File: main.c [hunk 2]
```diff
@@ -282,6 +283,8 @@ usage()
 	printf("\tEnable a specific keyboard layout decode table.\n");
 	printf("-sdcard <sdcard.img>\n");
 	printf("\tSpecify SD card image (partition map + FAT32)\n");
+	printf("-mountdir <directory>\n");
+	printf("\tSpecify a directory to switch to once loaded.\n");
```
Comment (mist64) [inline main.c]: "Specify a directory for the host filesystem bridge (device 1)."

**Comment by mist64** [comment]
* Good feature, thanks! Can we call it just `-dir` though?
* Two small comments in the diff.
* Also, CI failed because the compiler on Linux (GCC) insists that you check the return value of `chdir()`. You're already checking whether it exists, do just say `(void)chdir(mountdir_path);`.

**Comment by greg-king5** [comment]
That patch isn't needed.  We already have that feature.  We just do it the other way:

1. Change to the Directory where we have our programs.
2. Use a path-name to choose which emulator that we want to execute.

**Comment by izb** [comment]
> That patch isn't needed. We already have that feature. We just do it the other way:
> 
> 1. Change to the Directory where we have our programs.
> 2. Use a path-name to choose which emulator that we want to execute.

The emulator looks in its cwd to find the rom file, so this patch allows you to have your programs in a different location to the rom.

**Comment by greg-king5** [comment]
> > That patch isn't needed. We already have that feature. We just do it the other way:
> > 
> > 1. Change to the Directory where we have our programs.
> > 2. Use a path-name to choose which emulator that we want to execute.
> 
> The emulator looks in its cwd to find the rom file, so this patch allows you to have your programs in a different location to the rom.

[Not anymore -- it was fixed!](https://github.com/commanderx16/x16-emulator/releases/tag/r31)  The emulator now looks in the directory from which its binary executable was loaded.

I use the feature all the time.

**Comment by izb** [comment]
> > > That patch isn't needed. We already have that feature. We just do it the other way:
> > > 
> > > 1. Change to the Directory where we have our programs.
> > > 2. Use a path-name to choose which emulator that we want to execute.
> > 
> > 
> > The emulator looks in its cwd to find the rom file, so this patch allows you to have your programs in a different location to the rom.
> 
> [Not anymore -- it was fixed!](https://github.com/commanderx16/x16-emulator/releases/tag/r31) The emulator now looks in the directory from which its binary executable was loaded.
> 
> I use the feature all the time.

So it was! I'll close this PR

**Pull request closed without merging**